internal/xds/v3/callbacks.go

```diff
@@ -0,0 +1,47 @@
+// Copyright Project Contour Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v3
+
+import (
+	"fmt"
+
+	envoy_service_discovery_v3 "github.com/envoyproxy/go-control-plane/envoy/service/discovery/v3"
+	envoy_server_v3 "github.com/envoyproxy/go-control-plane/pkg/server/v3"
+	"github.com/sirupsen/logrus"
+)
+
+func NewCallbacks(log *logrus.Logger) envoy_server_v3.Callbacks {
+	return &envoy_server_v3.CallbackFuncs{
+		StreamRequestFunc: func(streamID int64, req *envoy_service_discovery_v3.DiscoveryRequest) error {
+			logDiscoveryRequestDetails(log, req)
+			return nil
+		},
+	}
+}
+
+func logDiscoveryRequestDetails(l *logrus.Logger, req *envoy_service_discovery_v3.DiscoveryRequest) {
+	log := l.WithField("version_info", req.VersionInfo).WithField("response_nonce", req.ResponseNonce)
+	if req.Node != nil {
```

Comment:
It would be helpful to log the node's name as well which today should correlate to the pod name.

Reply:
Does `node_id` capture what youre looking for? I think that is set as the pod name at least in a basic/example deployment by the bootstrapper https://github.com/projectcontour/contour/blob/e4ce0d18261c1c036674f9530ae2b0b08d4f4859/examples/contour/03-envoy.yaml#L52

```
time="2020-12-03T21:00:19Z" level=info msg="handling v3 xDS resource request" node_id=envoy-v2g4n node_cluster=projectcontour node_version=v1.16.1 resource_names="[default/kuard]" response_nonce=18 type_url=type.googleapis.com/envoy.config.endpoint.v3.ClusterLoadAssignment version_info=18
```

(I added `node_cluster` just now but not sure how useful that is?)

Reply:
I think `node_id` should cover it, unless @stevesloka had something else in mind?